File: frontend/src/api/categoryApi.ts

import { ApiCategoryResponse, Category } from '../types/Category.ts';
import { Product } from '../types/Product';
import axiosClient from './axiosClient.ts';

export const CategoryApi = {
    getCategories: async (): Promise<Category[]> => {
    return await axiosClient.get('/categories');
  },
    // Lấy chi tiết 1 category theo id hoặc slug
  getCategoryDetail: async (idOrSlug: string): Promise<Category> => {
    return await axiosClient.get(`/categories/${idOrSlug}`);
  },

  // Lấy sản phẩm theo category (slug)
  getProductsByCategoryPagination: async (
    slug: string,
    page: number = 1,
    limit: number = 6
  ): Promise<{ data: Product[]; pagination: { totalPages: number } }> => {
    return await axiosClient.get(
      `/products?category=${slug}&page=${page}&limit=${limit}`
    );
  },

  addCategory: async (data: Partial<Category>): Promise<ApiCategoryResponse<Category>> => {
    return await axiosClient.post("/admin/categories", data);
  },

}

File: Mobileptit/app/src/main/java/com/uteshop/app/data/remote/api/CategoryApi.java
package com.uteshop.app.data.remote.api;

import com.uteshop.app.data.model.Category;
import java.util.List;
import retrofit2.Call;
import retrofit2.http.*;

public interface CategoryApi {
    @GET("v1/api/categories")
    Call<List<Category>> getCategories();

    @GET("v1/api/categories/{slug}-{id}")
    Call<Category> getCategoryDetail(@Path("slug") String slug, @Path("id") String id);
}
